src/components/PopupWithForm.js

```diff
@@ -1,11 +1,21 @@
 import Popup from "./Popup.js";
 
 class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, buttonText, loadingButtonText) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._button = this._popupElement.querySelector(".modal__button");
+    this._button = this._popupElement.querySelector(".modal__button-submit");
     this._handleFormSubmit = handleFormSubmit;
+    this._buttonText = buttonText;
+    this._loadingButtonText = loadingButtonText;
+  }
+
+  showLoading() {
+    this._button.textContent = this._loadingButtonText;
+  }
+
+  hideLoading() {
+    this._button.textContent = this._buttonText;
   }
 
   _getInputValues() {
```

Comment:
```js

    const inputList = Array.from(
      this._popupElement.querySelectorAll(".modal__input")
    );
```

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.